client/hooks/useAuth.ts

```diff
@@ -3,30 +3,52 @@
  * @returns {object} - An object containing:
  *   - `isAuthenticated`: Boolean indicating whether the user is currently authenticated.
  *   - `isLoading`: Boolean indicating if the authentication status check is in progress.
+ *   - `signOut`: A  mutationfunction to sign out the user.
  */
 
 import { auth } from '@/firebase/firebaseConfig';
 import { useState, useEffect } from 'react';
-import { onAuthStateChanged } from 'firebase/auth';
+import { onAuthStateChanged, signOut } from 'firebase/auth';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import type { FirebaseError } from 'firebase/app';
+import useFetchUserProfile from './useFetchUserProfile';
 
 const useAuth = () => {
   const [isAuthenticated, setIsAuthenticated] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
+  const queryClient = useQueryClient();
+  const { refetch } = useFetchUserProfile();
+
+  const signOutMutation = useMutation<void, FirebaseError, void>({
```

Comment:
thank you for making our code more concise by joining the two auth hooks into one + a mutation